src/argilla/server/errors/api_errors.py

```diff
@@ -41,10 +43,20 @@ def __init__(self, error: ServerError):
 
 
 class APIErrorHandler:
+    @staticmethod
+    async def track_error(error: ServerError, request: Request):
+        data = {"code": error.code}
```

Comment:
We can setup the headers here instead of delegating a private telemetry method. This logic has more sense here.
```suggestion
        data = {
           "code": error.code,
           "user-agent": request.headers.get("user-agent"),
           "accept-language": request.headers.get("accept-language")
        }
```